hypershift-operator/main.go

```diff
@@ -326,25 +325,6 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 		}
 	}
 
-	// Since we dropped the validation webhook server we need to ensure this resource doesn't exist
-	// otherwise it will intercept kas requests and fail.
-	// TODO (alberto): dropped in 4.14.
-	validatingWebhookConfiguration := &admissionregistrationv1.ValidatingWebhookConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "ValidatingWebhookConfiguration",
-			APIVersion: admissionregistrationv1.SchemeGroupVersion.String(),
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: opts.Namespace,
-			Name:      hyperv1.GroupVersion.Group,
-		},
-	}
-	if err := mgr.GetClient().Delete(ctx, validatingWebhookConfiguration); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return err
-		}
-	}
-
```

Comment:
yes, we might have some envs running on versions where going through this is still necessary. So this would need to gate on if webhookNotEnabled kind of thing.
Also in addition please [consider using ignore fail policy](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy).

Reply:
Thanks @enxebre.

I'm not sure how to handle this. The flag is only known to the cli, and this is where the webhook is created. Main is running inside the cluster, knowing nothing about the installation and its configuration, AFAIK. 

The only thing I can think of, is to use different webhook name for the new one. I don't like this idea and I find it confusing and error prompt.  

Do you have any other idea?

Reply:
we can explore either move the logic out to "install" or propagate the input into the HO options